django_declarative_apis/machinery/errors.py

```diff
@@ -179,42 +182,44 @@ class ClientErrorExtraFields(ClientError):
     def __init__(self, extra_fields=None):
         error_code, error_message = EXTRA_FIELDS
         if extra_fields:
-            error_message += ": %s" % ", ".join(extra_fields)
+            error_message += f": {', '.join(extra_fields)}"
         super().__init__(error_code, error_message)
 
 
 class ClientErrorReadOnlyFields(ClientError):
     def __init__(self, read_only_fields=None):
         error_code, error_message = READ_ONLY_FIELDS
         if read_only_fields:
-            error_message += ": %s" % ", ".join(read_only_fields)
+            error_message += f": {', '.join(read_only_fields)}"
         super().__init__(error_code, error_message)
 
 
 class ClientErrorMissingFields(ClientError):
     def __init__(self, missing_fields=None, extra_message=None):
         error_code, error_message = MISSING_FIELDS
         if missing_fields:
-            error_message += ": %s" % ", ".join(missing_fields)
+            error_message += f": {', '.join(missing_fields)}"
         if extra_message:
-            error_message += ": {0}".format(extra_message)
+            error_message += f": {extra_message}"
         super().__init__(error_code, error_message)
 
 
 class ClientErrorInvalidFieldValues(ClientError):
     def __init__(self, invalid_fields=None, extra_message=None):
         error_code, error_message = INVALID_FIELD_VALUES
         if invalid_fields:
-            error_message += ": %s" % ", ".join(invalid_fields)
+            error_message += f": {', '.join(invalid_fields)}"
         if extra_message:
-            error_message += ": {0}".format(extra_message)
+            error_message += f": {extra_message}"
         super().__init__(error_code, error_message)
 
 
 class ServerError(ApiError):
-    def __init__(self):
+    def __init__(self, additional_info=None):
         error_code, error_message = LOGGED_SERVER_ERROR
         error_message = error_message.format(uuid.uuid4())
+        if additional_info:
+            error_message += f": {additional_info}"
```

Comment:
I believe that this `error_message` is returned directly to the client.  Should we make it so that the `additional_info` only appears in logs, and the client only gets the opaque error UUID?